src/test/groovy/com/github/benmanes/gradle/versions/DependencyUpdatesSpec.groovy

```diff
@@ -120,24 +121,10 @@ final class DependencyUpdatesSpec extends Specification {
     reporter.write()
 
     then:
-    reporter.unresolved.collect { it.selector }.collectEntries { dependency ->
-      [['group': dependency.group, 'name': dependency.name]: dependency.version]
-    } == [
-      ['group': 'com.github.ben-manes', 'name': 'unresolvable'] : '+',
-      ['group': 'com.github.ben-manes', 'name': 'unresolvable2']: '+',
-    ]
-    reporter.upgradeVersions == [
-      ['group': 'com.google.inject', 'name': 'guice']                         : '2.0',
-      ['group': 'com.google.inject.extensions', 'name': 'guice-multibindings']: '2.0',
-    ]
-    reporter.upToDateVersions == [
-      ['group': 'backport-util-concurrent', 'name': 'backport-util-concurrent']       : '3.1',
-      ['group': 'backport-util-concurrent', 'name': 'backport-util-concurrent-java12']: '3.1',
-    ]
-    reporter.downgradeVersions == [
-      ['group': 'com.google.guava', 'name': 'guava']      : '99.0-SNAPSHOT',
-      ['group': 'com.google.guava', 'name': 'guava-tests']: '99.0-SNAPSHOT',
-    ]
+    checkUnresolvedVersions(reporter)
+    checkUpgradeVersions(reporter)
+    checkUpToDateVersions(reporter)
+    checkDowngradeVersions(reporter)
```

Comment:
Because these collections are now objects, doing a direct comparisons of the collections was no longer practical. I revised these tests to instead check the size of the collection and individual entries using the appropriate keys.

I also abstracted them, rather that update one and copy/paste it five times.